### PR TITLE
Es6 projects common modules crosswords anagram helper ring

### PR DESCRIPTION
**Pull request by MatthewJWalls**
## What does this change?

ES6ify anagram helper ring for crosswords

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/crosswords/anagram-helper/ring.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            this.props.letters.map((letter, i) =>
+                React.createElement(
+                    'div',
+                    {
+                        className: `crossword__anagram-helper-shuffler__letter ${letter.entered
+                            ? 'entered'
+                            : ''}`,
+                        style: this.getPosition(angle, i),
+                        key: `l${i}`, // eslint-disable-line react/no-array-index-key
+                    },
```
Comment (MatthewJWalls) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/ring.js]: I wasn't sure what to do here for this eslint warning so open to advice. It's an array of characters, potentially duplicated.

Comment (SiAdcock) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/ring.js]: I think  the most succinct way to uniquely identify a letter without using its array index is by taking a combination of its left and top position. e.g.:

```js
const { left, right } = this.getPosition(angle, i);
React.createElement(
  'div', 
  {
    key: `${top},${left}`
  },
  letter.value
)
```
What do you think?

Comment (MatthewJWalls) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/ring.js]: ok

I'll give it a shot!

Comment (desbo) [inline static/src/javascripts/projects/common/modules/crosswords/anagram-helper/ring.js]: I can't remember if the `letters` in the props here have `x` and `y` coordinates (some letter objects definitely do), but that seems like a good way to key letters in general

**Comment by SiAdcock** [comment]
Closing this as it was completed in #16866

**Pull request closed without merging**